src/middleware/attachRedisClient.ts

```diff
@@ -44,6 +49,6 @@ export default async (req, res, next) => {
   res.on('finish', afterResponse);
   res.on('close', afterResponse);
 
-  req.redisClient = redisClient;
+  req.redisClient = redisClient as RedisClientType;
```

Comment:
rather than doing `as RedisClientType` here we want to annotate the `redisClient` when it's getting created on line 20

so we should do `const redisClient: RedisClientType = ...` if that doesn't work then you can cast the `redisClient` by using `as RedisClientType` on line 41.

```suggestion
  req.redisClient = redisClient;
```

Reply:
Yes, your suggestion is the right way to do it but it does not work "When Typescript is strict: true" because of some issues with RedisClientType. You can read [here](https://github.com/redis/node-redis/issues/1865). But will make the change

Reply:
makes sense, thanks for the link